src/main/java/com/searchgears/queryanalysis/matching/DictionaryRewriter.java

```diff
@@ -22,15 +19,28 @@
  * after applying tokenization and lowercasing.
  * It returns the normalized input text with all matching tokens replaced.
  */
-public class DictionaryRewriter implements ResourceLoaderAware {
+public class DictionaryRewriter {
     private final SynonymGraphFilterFactory synonymGraphFilterFactory;
 
-    DictionaryRewriter(String synFile) {
+    DictionaryRewriter(String synFile, ResourceLoader resourceLoader) {
+        synonymGraphFilterFactory = new SynonymGraphFilterFactory(getSynonymGraphSettings(synFile));
+        initSynonymGraphFilter(resourceLoader);
```

Comment:
I wouldn't call it "init" because that is a different lifecycle step in solr components. Rather stick with inform

Reply:
Rename init to inform